functorch/_src/vmap.py

```diff
@@ -22,6 +23,23 @@
 in_dims_t = Union[int, Tuple]
 out_dims_t = Union[int, Tuple[int, ...]]
 
+
+def register_namedtuple_return_types():
+    # Register torch.return_types as pytree node.
+    for name in dir(torch.return_types):
+        if name.startswith('__'):
+            continue
+        attr = getattr(torch.return_types, name)
+        if inspect.isclass(attr):
+            return_type_class = attr
+            # Note: We capture the current `return_type_class` with default argument `constructor`
+            # in the lambda otherwise we will point to the last value of `return_type_class` for all lambdas
+            torch.utils._pytree._register_pytree_node(return_type_class, lambda x: (
+                tuple(x), None), lambda x, c, constructor=return_type_class: constructor(x))
```

Comment:
tuple(x) should be list(x): flatten_function are supposed to return List[Any]: https://github.com/pytorch/pytorch/blob/de902b5d02ef0b264d4ce1406b8e16c7ba1e2680/torch/utils/_pytree.py#L54-L55